mlflow/java/src/main/java/org/mlflow/sagemaker/MLeapPredictor.java

```diff
@@ -1,18 +1,109 @@
 public class MLeapPredictor extends Predictor {
-  public MLeapPredictor(String modelDataPath, String inputSchemaPath) {}
+  private final Transformer pipelineTransformer;
+  private final LeapFrameSchema inputSchema;
+
+  private static final Seq<String> predictionColumnNames;
+  private static final Logger logger = LoggerFactory.getLogger(MLeapPredictor.class);
+
+  static {
+    List<String> predictionColumnList = Arrays.asList("prediction");
+    predictionColumnNames =
+        JavaConverters.asScalaIteratorConverter(predictionColumnList.iterator()).asScala().toSeq();
+  }
+
+  /**
+   * Constructs an {@link MLeapPredictor}
+   *
+   * @param modelDataPath The path to the serialized MLeap model
+   * @param inputSchema The path to JSON-formatted file containing the input schema that the model
+   *     accepts
+   */
+  public MLeapPredictor(String modelDataPath, String inputSchemaPath) {
+    MleapContext mleapContext = new ContextBuilder().createMleapContext();
+    BundleBuilder bundleBuilder = new BundleBuilder();
+    this.pipelineTransformer = bundleBuilder.load(new File(modelDataPath), mleapContext).root();
+    try {
+      this.inputSchema = LeapFrameSchema.fromPath(inputSchemaPath);
+    } catch (IOException e) {
+      logger.error("Could not read the model input schema from the specified path", e);
+      throw new PredictorLoadingException(String.format(
+          "Failed to load model input schema from specified path: %s", inputSchemaPath));
+    }
+  }
 
   @Override
-  protected PredictorDataWrapper predict(PredictorDataWrapper input) {
-    throw new UnsupportedOperationException("Not yet implemented!");
+  protected PredictorDataWrapper predict(PredictorDataWrapper input)
+      throws PredictorEvaluationException {
+    PandasRecordOrientedDataFrame pandasFrame = null;
+    try {
+      pandasFrame = PandasRecordOrientedDataFrame.fromJson(input.toJson());
+    } catch (IOException e) {
+      logger.error(
+          "Encountered a JSON conversion error during conversion of Pandas dataframe to LeapFrame.",
+          e);
+      throw new PredictorEvaluationException(
+          "Failed to transform input into a JSON representation of an MLeap dataframe."
+          + "Please ensure that the input is a JSON-serialized Pandas Dataframe"
+          + "with the `record` orientation");
+    }
+
+    DefaultLeapFrame leapFrame = null;
+    try {
+      leapFrame = pandasFrame.toLeapFrame(this.inputSchema);
+    } catch (InvalidSchemaException e) {
+      throw new PredictorEvaluationException(
+          "Encountered a schema mismatch when converting the input dataframe to a LeapFrame.");
+    } catch (Exception e) {
+      logger.error(
+          "Encountered an unknown error during conversion of Pandas dataframe to LeapFrame.", e);
+      throw new PredictorEvaluationException(
+          "An unknown error occurred while converting the input dataframe to a LeapFrame.");
```

Comment:
let's include the original exception `e` here.

Reply:
Added the output of `e.GetMessage()` to the new exception text.